invenio_github/models.py

```diff
@@ -129,6 +262,54 @@ class Release(db.Model, Timestamp):
     )
     """Record identifier."""
 
-    repository = db.relationship(Repository, backref='releases')
+    status = db.Column(
+        ChoiceType(ReleaseStatus, impl=db.CHAR(1)),
+        nullable=False,
+    )
+    """Status of the release, e.g. 'processing', 'published', 'failed', etc."""
+
+    repository = db.relationship(
+        Repository,
+        backref=db.backref('releases', lazy='dynamic')
+    )
     record = db.relationship(RecordMetadata)
     event = db.relationship(Event)
+
+    @classmethod
+    def create(cls, event):
+        release_id = event.payload['release']['id']
+
+        # Check if the release has already been received
+        existing_release = Release.query.filter_by(
+            release_id=release_id,
+        ).first()
+        if existing_release:
+            msg = 'Release [{0}:{1}] has already been received with status {1}'
```

Comment:
What about using names instead of numbers for string formatting?

```python
'Release [{tag}:{release_id}] ...: {status}.'.format(tag=..., release_id=..., ...)
```